app/views/HomePage/index.tsx

```diff
@@ -105,10 +101,24 @@ function BookItem(props: BookProps) {
 }
 
 function HomePage() {
+    const [page, setPage] = useState<number>(1);
+
+    const { data: result, loading, refetch } = useQuery(FEATURED_BOOKS, {
+        variables: {
+            page: 1,
+            pageSize: 10,
+        },
+    });
+    const books = (!loading && result) ? result.books.results : [];
     const bookItemRendererParams = React.useCallback((_, data) => ({
         book: data,
     }), []);
 
+    const nextPage = React.useCallback(() => {
+        setPage(page + 1);
+        refetch();
+    }, []);
```

Comment:
1. useCallback does not have correct dependencies.
2. Please use `setPage((oldPage) => oldPage + 1)` syntax to avoid adding dependency